splink/cluster_metrics.py

```diff
@@ -1,53 +1,123 @@
+from typing import Dict, List
+
 from splink.splink_dataframe import SplinkDataFrame
 
 
-def _size_density_sql(
+def _node_degree_sql(
     df_predict: SplinkDataFrame,
     df_clustered: SplinkDataFrame,
-    threshold_match_probability: float,
     composite_uid_edges_l: str,
+    composite_uid_edges_r: str,
     composite_uid_clusters: str,
-):
-    """Generates sql for computing cluster size and density at a given threshold.
+    threshold_match_probability: float,
+) -> List[Dict[str, str]]:
+    """
+    Generates sql for computing node degree per node, at a given edge threshold.
+
+    This is includes nodes with no edges, as identified via the clusters table.
 
     Args:
         df_predict (SplinkDataFrame): The results of `linker.predict()`.
         df_clustered (SplinkDataFrame): The outputs of
                 `linker.cluster_pairwise_predictions_at_threshold()`.
+        composite_uid_edges_l (str): unique id for left-hand edges.
+        composite_uid_edges_r (str): unique id for right-hand edges.
+        composite_uid_clusters (str): unique id for clusters.
         threshold_match_probability (float): Filter the pairwise match
             predictions to include only pairwise comparisons with a
             match_probability above this threshold.
```

Comment:
```suggestion
            match_probability at or above this threshold.
```
My boo-boo from before